src/storage/sftp/storage.c

```diff
@@ -897,6 +1146,140 @@ storageSftpNew(
+                // empty known hosts file will return 0.
+                if ((rc = libssh2_knownhost_readfile(knownHostsList, currentKnownHostFile, LIBSSH2_KNOWNHOST_FILE_OPENSSH)) <= 0)
+                {
+                    if (rc == 0)
+                        LOG_DETAIL_FMT("libssh2 '%s' file is empty", currentKnownHostFile);
+                    else
+                    {
+                        char *libSsh2ErrMsg;
+                        int libSsh2ErrMsgLen;
+
+                        // Get the libssh2 error message
+                        rc = libssh2_session_last_error(this->session, &libSsh2ErrMsg, &libSsh2ErrMsgLen, 0);
+
+                        LOG_DETAIL_FMT(
+                            "libssh2 read '%s' failed: libssh2 errno [%d] %s", currentKnownHostFile, rc, libSsh2ErrMsg);
+                    }
+                }
+                else
+                    LOG_DETAIL_FMT("libssh2 read '%s' succeeded", currentKnownHostFile);
+            }
+
+            // Get the remote host key
+            hostKey = libssh2_session_hostkey(this->session, &hostKeyLen, &hostKeyType);
+
+            // Check for a match in known hosts files else throw an error if no host key was retrieved
+            if (hostKey != NULL)
+            {
+                rc = libssh2_knownhost_checkp(
+                    knownHostsList, strZ(host), (int)port, hostKey, hostKeyLen,
+                    LIBSSH2_KNOWNHOST_TYPE_PLAIN | LIBSSH2_KNOWNHOST_KEYENC_RAW, NULL);
+
+                // Handle check success/failure
+                if (rc == LIBSSH2_KNOWNHOST_CHECK_MATCH)
+                    LOG_DETAIL_FMT("known hosts match found for '%s'", strZ(host));
+                else
+                {
+                    // We did not get a match, get an appropriate failure message
+                    const char *const matchFailMsg = storageSftpKnownHostCheckpFailureMsg(rc);
+
+                    // Handle failure to match in the same manner as ssh_config StrictHostKeyChecking for yes/accept-new/no/off.
+                    // If this flag is set to yes, never automatically add host keys to the ~/.ssh/known_hosts file, and refuse to
+                    // connect to hosts whose host key has changed. This option forces the user to manually add all new hosts. If
+                    // this flag is set to “accept-new” then automatically add new host keys to the user known hosts files, but do
+                    // not permit connections to hosts with changed host keys. If this flag is set to “no” or “off”, automatically
+                    // add new host keys to the user known hosts files and allow connections to hosts with changed hostkeys to
+                    // proceed.
+                    switch (param.sftpStrictHostKeyChecking)
+                    {
+                        case SFTP_STRICT_HOSTKEY_CHECKING_YES:
+
```

Comment:
All of these cases should get braces. See https://github.com/pgbackrest/pgbackrest/blob/main/CODING.md#braces

Reply:
Updates in
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/4fd017c80716d71a7ffff1f25379b9da196f7001
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/6811405f150c8f1095c3da31d952f0ddebc58be1
https://github.com/pgbackrest/pgbackrest/pull/2125/commits/a3a82046feab0a915e0c116f867818b86ae42621